quark/foreach.py

```diff
@@ -83,19 +71,35 @@ def run():
         # the current working directory to the submodules root directory
         os.environ["displaypath"] = str(module_displaypath)
 
-        # sha1 is the commit as recorded in the immediate superproject
-        os.environ["sha1"] = str(commit_sha)
-
         # toplevel is the absolute path to the
         # top-level of the immediate superproject
         os.environ["toplevel"] = str(toplevel)
 
+        # version_control is the version control used for the subproject
+        os.environ["version_control"] = version_control
+
+        if version_control == "git":
+            # Case for git
+            # sha1 is the commit of the subproject ( empty string
+            # if it is a svn repository )
+            os.environ["sha1"] = str(module.ref)
+        else:
+            # Case for SVN
+            # rev is the revision of the subproject ( empty string if
+            # it is a git repository )
+            os.environ["rev"] = module.rev
+
         with DirectoryContext(module.directory):
             output = subprocess.check_output(cmd, shell=True)
 
             if not optlist.quiet:
                 print(output.decode("utf-8"))
 
+        try:
+            del os.environ["sha1"]
```

Comment:
I don't really like that we are polluting the environment of our own process, and then we have to do these tricks to restore it... A cleaner and better alternative would be to populate a dictionary (starting with `dict(os.environ)`), pass it as the `env` argument of `check_output` and throw it away at each iteration.

Reply:
I really like your solution and I added it in https://github.com/comelz/quark/pull/27/commits/bec2dd5ceb5226d28cddc00dab60771514ea94bd